website/docs/components/table/partials/code/how-to-use.md

```diff
@@ -545,6 +545,153 @@ In this case the table layout is still set to `auto` (default). If instead you w
+
+!!!
+
+#### Multi-select table with pagination and persisted selection status
+
+This is a more complex example, where a table with multi-selection is associated with a [Pagination](/components/pagination) element (a similar use case would apply if a [filter](/patterns/filter-patterns) is applied to the data used to populate the table). In this case, a **subset of rows** is displayed on screen.
+
+When a user selects a row, if the displayed rows are replaced with other ones (e.g., when the user clicks on the "next" button or on a different page number) there's the question of what happens to the previous selection: is it persisted in the data/model underlying the table? Or is it lost?
+
+In the demo below we are persisting the selection in the data/model, so that when navigating to different pages the row selections are persisted across table re-renderings.
+
+```handlebars
+<div class="doc-table-multiselect-with-pagination-demo">
+  <Hds::Table
+    @isSelectable={{true}}
+    @onSelectionChange={{this.demoOnSelectionChangeWithPagination}}
+    @model={{this.demoPaginatedData}}
+    @columns={{array
+      (hash key="artist" label="Artist")
+      (hash key="album" label="Album")
+      (hash key="year" label="Year")
+    }}
+  >
+    <:body as |B|>
+      <B.Tr @selectionKey={{B.data.id}} @isSelected={{B.data.isSelected}} @selectionAriaLabelSuffix="row {{B.data.artist}} / {{B.data.album}}">
+        <B.Td>{{B.data.artist}}</B.Td>
+        <B.Td>{{B.data.album}}</B.Td>
+        <B.Td>{{B.data.year}}</B.Td>
+      </B.Tr>
+    </:body>
+  </Hds::Table>
+  <Hds::Pagination::Numbered
+    @totalItems={{this.demoTotalItems}}
+    @currentPage={{this.demoCurrentPage}}
+    @pageSizes={{array 2 4}}
+    @currentPageSize={{this.demoCurrentPageSize}}
+    @onPageChange={{this.demoOnPageChange}}
+    @onPageSizeChange={{this.demoOnPageSizeChange}}
+    @ariaLabel="Pagination for multi-select table"
+  />
+</div>
+```
+
+Depending on the expected behavior, you will need to implement the consumer-side logic that handles the persistence (or not) using the `@onSelectionChange` callback function. For the example above, something like this:
+
+```javascript
+@action
+demoOnSelectionChangeWithPagination({ selectableRowsStates }) {
+  // we loop over all the displayed table rows (a subset of the dataset)
+  selectableRowsStates.forEach((row) => {
+    // we find the record in the dataset corresponding to the current row
+    const recordToUpdate = this.demoSourceData.find(
+      (modelRow) => modelRow.id === row.selectionKey
+    );
+    if (recordToUpdate) {
+      // we update the record `isSelected` state based on the row (checkbox) state
+      recordToUpdate.isSelected = row.isSelected;
+    }
+  });
+}
+
+```
+
+For details about the arguments provided to the `@onSelectionChange` callback function refer to the [Component API](#component-api) section.
+
+
+#### Usability and accessibility considerations
```

Comment:
@hashicorp/hds-design please review this section (it may overlap with your content about this topic)